etc/spack/defaults/concretizer.yaml

```diff
@@ -10,6 +10,9 @@
 # or by adding a `concretizer:` section to an environment.
 # -------------------------------------------------------------------------
 concretizer:
+  # Maximum time, in seconds, allowed to clingo for the 'solve' phase. If set
+  # to 0, there is no time limit.
+  timeout: 0
```

Comment:
should probably move all the way down, and explain what happens on timeout, and whether you'd ever want to set this.

Reply:
See https://github.com/spack/spack/pull/47661/commits/c47c38332d2e0c63e41d5301f09a3c7032bc3ef0

@sethrj The last commit should take also into consideration your concern on non-determinism. Non-deterministic behavior is now opt-in. By default we error on timeouts.